credentials/src/main/kotlin/SSI.kt

```diff
@@ -308,24 +307,40 @@ public object VerifiablePresentation {
     return JwtVerifiablePresentation.fromCompactSerialization(vpJWT)
       .verify_Ed25519_EdDSA(publicKeyJWK.toOctetKeyPair())
   }
-}
 
-/**
- * The selectFrom method is a helper function that helps filter out the verifiable credentials which can not be selected and returns
- * the selectable credentials.
- *
- * @param presentationDefinition definition of what is expected in the presentation.
- * @param vcJwts verifiable credentials are the credentials from wallet provided to the library to find selectable credentials.
- *
- * @return the selectable credentials.
- */
-@Throws(Exception::class)
-private fun selectFrom(presentationDefinition: PresentationDefinitionV2, vcJwts: List<VcJwt>): List<VcJwt> {
-  val selectableCredentials = mutableListOf<VcJwt>()
+  /**
+   * Decodes a verifiable presentation JWT into its header, payload, and signature components.
+   *
+   * @param vpJWT The JWT representation of the verifiable presentation to be decoded.
+   * @return A [DecodedJwt] object containing the decoded components.
+   */
+  public fun decode(vpJWT: VpJwt): DecodedJwt {
+    val (encodedHeader, encodedPayload, encodedSignature) = vpJWT.split('.')
+
+    return DecodedJwt(
+      header = String(Base64.getDecoder().decode(encodedHeader)),
+      payload = String(Base64.getDecoder().decode(encodedPayload)),
+      signature = encodedSignature
+    )
```

Comment:
Nimbus already has this. Is it possible to remove the `DecodedJwt`? Why is it part of our API? See https://connect2id.com/products/nimbus-jose-jwt/examples/jose-jwt-parsing

Reply:
Removed